app/data_models/questionnaire_store.py

```diff
@@ -60,30 +63,54 @@ def set_metadata(self, to_set: MutableMapping) -> QuestionnaireStore:
 
         return self
 
-    def set_supplementary_data(self, to_set: MutableMapping) -> None:
+    def set_supplementary_data(
+        self,
+        *,
+        to_set: MutableMapping,
+        base_questionnaire_store_updater: BaseQuestionnaireStoreUpdater,
+    ) -> None:
         """
         Used to set or update the supplementary data whenever the sds endpoint is called
         (Which should be once per session, but only if the sds_dataset_id has changed)
 
         this updates ListStore to add/update any lists for supplementary data and stores the
         identifier -> list_item_id mappings in the supplementary data store to use in the payload at the end
         """
+        modified_lists: set[str] = set()
+
         if self._stores.supplementary_data_store.list_mappings:
-            self._remove_old_supplementary_lists_and_answers(new_data=to_set)
+            self._remove_old_supplementary_lists_and_answers(
+                new_data=to_set,
+                base_questionnaire_store_updater=base_questionnaire_store_updater,
+                modified_lists=modified_lists,
```

Comment:
This is usually bad practice. Avoid mutation of arguments passed in. It would make be more accepted if it was needed across multiple methods, given modified lists are not needed external to this method, I suggest the methods returning the modified lists.

Reply:
Makes sense, switched to returning them, thanks